File: tests/expected/loop.jl
function for_with_break()
    for i = 0:4-1
        if i == 2
            break
        end
        println(join([i], " "))
    end
end

function for_with_continue()
    for i = 0:4-1
        if i == 2
            continue
        end
        println(join([i], " "))
    end
end

function for_with_else()
    has_break = false
    for i = 0:4-1
        println(join([i], " "))
    end
    if has_break != true
        println(join(["OK"], " "))
    end
end

function while_with_break()
    i = 0
    while true
        if i == 2
            break
        end
        println(join([i], " "))
        i += 1
    end
end

function while_with_continue()
    i = 0
    while i < 5
        i += 1
        if i == 2
            continue
        end
        println(join([i], " "))
    end
end

function main()
    for_with_break()
    for_with_continue()
    for_with_else()
    while_with_break()
    while_with_continue()
end

main()

File: tests/expected/loop.go
package main

import (
	"fmt"
	iter "github.com/hgfischer/go-iter"
)

func ForWithBreak() {
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(4)).All() {
		if i == 2 {
			break
		}
		fmt.Printf("%v\n", i)
	}
}

func ForWithContinue() {
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(4)).All() {
		if i == 2 {
			continue
		}
		fmt.Printf("%v\n", i)
	}
}

func ForWithElse() {
	var has_break bool = false
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(4)).All() {
		fmt.Printf("%v\n", i)
	}
	if has_break != true {
		fmt.Printf("%v\n", "OK")
	}
}

func WhileWithBreak() {
	var i int = 0
	for true {
		if i == 2 {
			break
		}
		fmt.Printf("%v\n", i)
		i += 1
	}
}

func WhileWithContinue() {
	var i int = 0
	for i < 5 {
		i += 1
		if i == 2 {
			continue
		}
		fmt.Printf("%v\n", i)
	}
}

func main() {
	ForWithBreak()
	ForWithContinue()
	ForWithElse()
	WhileWithBreak()
	WhileWithContinue()
}
